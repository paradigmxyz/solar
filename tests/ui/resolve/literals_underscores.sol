contract LT {
    uint constant _TLOAD_TEST_PAYLOAD1 = 0x6002_601e_613d5c_3d_52_f3;
    uint constant _TLOAD_TEST_PAYLOAD2 = 0x6002_601E_613d5c_3d_52_f3;

    uint a = 1000_; //~ ERROR: invalid use of underscores in number literal
    uint b = 100__0; //~ ERROR: invalid use of underscores in number literal
    uint c = 1_.4e10; //~ ERROR: invalid use of underscores in number literal
    uint d = 3.4_e10; //~ ERROR: invalid use of underscores in number literal
    uint e = 3.4e_10; //~ ERROR: invalid use of underscores in number literal

    uint g = 1_.4e10 + 3.4e_10; //~ ERROR: invalid use of underscores in number literal
    //~^ ERROR: invalid use of underscores in number literal

    uint X1 = 0x1234__1234__1234__123; //~ ERROR: invalid use of underscores in number literal

    uint D1 = 1234_; //~ ERROR: invalid use of underscores in number literal
    uint D2 = 12__34; //~ ERROR: invalid use of underscores in number literal
    uint D3 = 12_e34; //~ ERROR: invalid use of underscores in number literal
    uint D4 = 12e_34; //~ ERROR: invalid use of underscores in number literal

    uint F1 = 3.1415_; //~ ERROR: invalid use of underscores in number literal
    uint F2 = 3__1.4__15; //~ ERROR: invalid use of underscores in number literal
    uint F3 = 1_.2; //~ ERROR: invalid use of underscores in number literal
    uint F4 = 1._2; //~ ERROR: invalid use of underscores in number literal
    uint F5 = 1.2e_12; //~ ERROR: invalid use of underscores in number literal
    uint F6 = 1._; //~ ERROR: empty rational
}
