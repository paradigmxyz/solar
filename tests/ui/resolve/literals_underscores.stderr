error: empty rational
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint F6 = 1._;
   |               ^^^
   |

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint a = 1000_;
   |              ^^^^^
   |
   = help: remove trailing underscores

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint b = 100__0;
   |              ^^^^^^
   |
   = help: only 1 consecutive underscore `_` is allowed between digits

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint c = 1_.4e10;
   |              ^^^^^^^
   |
   = help: remove underscores in front of the fraction part

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint d = 3.4_e10;
   |              ^^^^^^^
   |
   = help: remove underscores at the end of the mantissa

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint e = 3.4e_10;
   |              ^^^^^^^
   |
   = help: remove underscores in front of the exponent

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint g = 1_.4e10 + 3.4e_10;
   |              ^^^^^^^
   |
   = help: remove underscores in front of the fraction part

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint g = 1_.4e10 + 3.4e_10;
   |                        ^^^^^^^
   |
   = help: remove underscores in front of the exponent

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint X1 = 0x1234__1234__1234__123;
   |               ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: only 1 consecutive underscore `_` is allowed between digits

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint D1 = 1234_;
   |               ^^^^^
   |
   = help: remove trailing underscores

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint D2 = 12__34;
   |               ^^^^^^
   |
   = help: only 1 consecutive underscore `_` is allowed between digits

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint D3 = 12_e34;
   |               ^^^^^^
   |
   = help: remove underscores at the end of the mantissa

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint D4 = 12e_34;
   |               ^^^^^^
   |
   = help: remove underscores in front of the exponent

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint F1 = 3.1415_;
   |               ^^^^^^^
   |
   = help: remove trailing underscores

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint F2 = 3__1.4__15;
   |               ^^^^^^^^^^
   |
   = help: only 1 consecutive underscore `_` is allowed between digits

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint F3 = 1_.2;
   |               ^^^^
   |
   = help: remove underscores in front of the fraction part

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint F4 = 1._2;
   |               ^^^^
   |
   = help: remove underscores in front of the fraction part

error: invalid use of underscores in number literal
  --> ROOT/tests/ui/resolve/literals_underscores.sol:LL:CC
   |
LL |     uint F5 = 1.2e_12;
   |               ^^^^^^^
   |
   = help: remove underscores in front of the exponent

error: aborting due to 18 previous errors

